framework/src/postprocessors/SideFluxAverage.C

```diff
@@ -9,8 +9,8 @@
 
 #include "SideFluxAverage.h"
 
-registerMooseObject("MooseApp", SideFluxAverage);
-registerMooseObject("MooseApp", ADSideFluxAverage);
+registerMooseObjectDeprecated("MooseApp", SideFluxAverage, "06/30/2021 24:00");
+registerMooseObjectDeprecated("MooseApp", ADSideFluxAverage, "06/30/2021 24:00");
```

Comment:
So this diff is too huge for a rename and now I see why. I think the registration macro you are looking for is `registerMooseObjectRenamed` which:
```
/// add a deprecated MooseObject orig_class to the registry that has been replaced by another
/// object new_class with the same API. time is the time the object became/becomes deprecated in
/// "mm/dd/yyyy hh:mm" format.
/// A call to registerMooseObject is still required for the new class
#define registerMooseObjectRenamed(app, orig_class, time, new_class)
```
You can do a `git mv` on these files to rename them, and then replace all the names in the class implementation, but we should not have two classes floating around that do the same thing.

Reply:
I'm hoping if you do this, the diff will be quite small (and then you can also ignore all my previous requested changes since you didn't actually add those lines)

Reply:
Ok I've done this. 
Since the header is gone, Bison will not build anymore. 
This is the Bison PR needed https://hpcgitlab.hpc.inl.gov/idaholab/bison/-/merge_requests/2554

So it ll go: merge moose, submodule update in the Bison PR, merge bison PR right?

Reply:
I think you can still make this compatible. I would add back `SideFluxIntegral.h` and in that file `#include "SideDiffusionFluxIntegral.h` or whatever the new name is and then either `typedef NewName OldName` or maybe even:
```
class NewName : public OldName
{
public:
NewName();

static InputParmeters validParams();
};
```
in the event that there are forward declarations of `class OldName`